BigN120MinimumCostTicketUsingDP.java

```diff
@@ -0,0 +1,35 @@
+//Time complexity is O(nlogn)
+//Space complexity is O(1)
+//This solution is submitted on leetcode
+
+import java.util.HashSet;
+
+public class BigN120MinimumCostTicketUsingDP {
+	class Solution {
+		public int mincostTickets(int[] days, int[] costs) {
+			// edge case
+			if (days == null || costs == null)
+				return 0;
+			if (days.length == 0 || costs.length == 0)
+				return 0;
+			HashSet<Integer> set = new HashSet<>();
+			for (int i = 0; i < days.length; i++) {
+				set.add(days[i]);
+			}
+			int[] dp = new int[366];
```

Comment:
Rather than an array of 366 we can take the array of size which is maximum element in days array